the-experts-BE/app.js

```diff
@@ -33,6 +39,47 @@ app.post("/addContractor", function(req, res, next) {
   );
 });
 
+app.post("/addReview", function(req, res, next) {
+  queries.addReview(req.body, (err, dataResponse) => {
+    if (err) next(err);
+    res.json(dataResponse);
+  });
+});
+
+app.post("/login", function(req, res, next) {
+  queries.getUser(req.body.email, (err, dataResponse) => {
+    if (err) next(err);
+    comparePasswords(req.body.password,dataResponse.password,(error,result)=>{
+      if(error) next(error);
+      if(!result) res.json(null);
+      const jwt=sign(req.body.email,SECRET);
+      res.cookie('jwt',jwt);
+    res.json(jwt);
+  });
+});
+});
+
+app.post("/signup", function(req, res, next) {
+  let body = {
+    first_name: req.body.first_name,
+    last_name: req.body.last_name,
+    email: req.body.email,
+    user_password: req.body.user_password
+  };
+  let jwt = req.cookies.jwt;
+  if (cookie) res.json(jwt);
+  hashPassword(body.user_password, (err, result) => {
+    if (err) next(err);
+    body.password = result;
+    queries.addUser(body, (err, dataResponse) => {
+      if (err) next(err);
+      jwt = sign(req.body.email, SECRET);
+      res.cookie("jwt", jwt);
+      res.json(jwt);
```

Comment:
If you are putting the jwt in a cookie, why do you need to send it back in res.json()? Remember that, once set, cookies get sent back to server automatically with every request.

Why not just set jwt cookie and send back helpful message to FE eg: res.send('you are logged in successfully')?